### PR TITLE
Fix promo code read when fields are null

### PR DESCRIPTION
**Pull request by masoudghaffarinia**
This is an attempt to fix https://github.com/lukasaron/terraform-provider-stripe/issues/73.

### CODE REVIEW

**Comment by masoudghaffarinia** [comment]
Not sure this is gonna fix the issue. Is there a document on how I can use this repo locally in my project?

**Comment by masoudghaffarinia** [comment]
I tested this locally with my own project and it works fine.

Screenshot 2024-07-09 at 11 54 33 AM

File: stripe/resource_stripe_promotion_code.go [hunk 1]
```diff
@@ -194,20 +205,42 @@ func resourceStripePromotionCodeRead(_ context.Context, d *schema.ResourceData,
...
+				}
+		
+				if len(restrictions) > 0 {
+					return d.Set("restrictions", []map[string]interface{}{restrictions})
+				}
+			}
+			return nil
+		}(),
+		// only set metadata if it's not empty map
+		func() error {
```
Comment (lukasaron) [inline stripe/resource_stripe_promotion_code.go:238]: Could you please remove this modification?
1. It's not relevant to the issue.
2. It's not consistent with the other resources.
Looks good as an improvement and I would rather see another PR where all resources follow the same pattern.

File: stripe/resource_stripe_promotion_code.go [hunk 2]
```diff
@@ -132,21 +132,32 @@ func resourceStripePromotionCodeCreate(ctx context.Context, d *schema.ResourceDa
...
+
+		restrictionsMap := ToMap(restrictions)
+
+		if v, set := restrictionsMap["first_time_transaction"]; set {
+			params.Restrictions.FirstTimeTransaction = stripe.Bool(ToBool(v))
+		}
+
+		if v, min_amount_set := restrictionsMap["minimum_amount"]; min_amount_set {
+			amount := ToInt64(v)
+			if amount >= 1 {
```
Comment (lukasaron) [inline stripe/resource_stripe_promotion_code.go:152]: `if amount >= 1` ~> `if amount > 0`

File: stripe/resource_stripe_promotion_code.go [hunk 3]
```diff
@@ -132,21 +132,32 @@ func resourceStripePromotionCodeCreate(ctx context.Context, d *schema.ResourceDa
 		if err != nil {
 			return diag.FromErr(err)
 		}
-		params.ExpiresAt = stripe.Int64(t.Unix())
+		expiresAtUnixStripeValue := stripe.Int64(t.Unix())
```
Comment (lukasaron) [inline stripe/resource_stripe_promotion_code.go:135]: I have a feeling this block can be rewritten by leveraging https://pkg.go.dev/time#Time.IsZero into:

`
if t.IsZero() {
  params.ExpiresAt = stripe.Int64(t.Unix())
}
`

Comment (lukasaron) [reply stripe/resource_stripe_promotion_code.go:135]: The formatting is scattered by GitHub (the if condition should be in the block not inline).

**Review by lukasaron — CHANGES_REQUESTED** [review]

File: stripe/resource_stripe_promotion_code.go (hunk 2 above)
Comment (masoudghaffarinia) [reply stripe/resource_stripe_promotion_code.go:152]: Done!

File: stripe/resource_stripe_promotion_code.go (hunk 1 above)
Comment (masoudghaffarinia) [reply stripe/resource_stripe_promotion_code.go:238]: Done!

File: stripe/resource_stripe_promotion_code.go (hunk 3 above)
Comment (masoudghaffarinia) [reply stripe/resource_stripe_promotion_code.go:135]: Done!

**Review by lukasaron — APPROVED** [review]

**Pull request merged**